web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -457,23 +439,60 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     marker.setIcon(newIcon);
   }
 
-  private selectPolygons(locationOfInterestId: string | null) {
-    const polygons = locationOfInterestId
-      ? this.polygons.get(locationOfInterestId)
-      : undefined;
-    if (polygons) {
-      for (const polygon of polygons) {
-        polygon.setOptions({strokeWeight: enlargedPolygonStrokeWeight});
+  private selectMarker(locationOfInterestId: string | null) {
+    if (locationOfInterestId) {
+      const marker = this.markers.get(locationOfInterestId);
```

Comment:
Please use short-circuit return to avoid code-nesting hadouken 💨 :

```ts
if (!locationofInterestId) return;
const marker = ...
```

Maybe what does `this.panAndZoom(marker?.getPosition())` do when `marker` is null or undefined?

Lastly, you can simplify further with:

```ts
marker.setDraggable(this.shouldEnableDrawingTools);
```

Reply:
there was a check if marker is null or not. so that question mark was useless. my bad.